internal/http/services/owncloud/ocdav/ocdav.go

```diff
@@ -226,6 +227,13 @@ func NewWith(conf *Config, fm favorite.Manager, ls LockSystem, _ *zerolog.Logger
 	if err := s.davHandler.init(conf); err != nil {
 		return nil, err
 	}
+	if gwc == nil {
+		var err error
+		s.gwClient, err = pool.GetGatewayServiceClient(s.c.GatewaySvc)
```

Comment:
How about moving this default behavior into `New` and just erroring out in here if the given client is `nil`?

Reply:
I would have to duplicate it to `pkg/micro/ocdav/service.go` as well, otherwise we would change the behavior of the micro ocdav service, requiring a major version bump. So I put the fallback here, to make passing in a gwclient optional.